src/main/scala/scalapb/json4s/JsonFormat.scala

```diff
@@ -385,13 +385,15 @@ class Parser private (config: Parser.ParserConfig) {
   }
 
   def defaultEnumParser(enumDescriptor: EnumDescriptor, value: JValue): EnumValueDescriptor = value match {
-    case JInt(v) => enumDescriptor.findValueByNumber(v.toInt).getOrElse(throw new JsonFormatException(s"Invalid enum value: ${v.toInt} for enum type: ${enumDescriptor.fullName}"))
-    case JString(s) =>
-      enumDescriptor.values.find(_.name == s).getOrElse(throw new JsonFormatException(s"Unrecognized enum value '${s}'"))
-    case _ =>
-      throw new JsonFormatException(
-        s"Unexpected value ($value) for enum ${enumDescriptor.fullName}")
-  }
+      case JInt(v) => enumDescriptor.findValueByNumber(v.toInt)
+          .getOrElse(enumDescriptor.findValueByNumberCreatingIfUnknown(0))
+      case JString(s) if config.isIgnoringUnknownFields => enumDescriptor.values.find(_.name == s)
+          .getOrElse(NullValue.NULL_VALUE.scalaValueDescriptor)
```

Comment:
This is probably what causing the unexpected test outcome. `NullValue` is unrelated to the arbitrary enum we have here. The tricky part is to find a way to bubble up `None` out of `defaultEnumParser` - maybe the signature needs to chage to return `Option[EnumValueDescriptor]`, and that would require a binary-incompatible change to [registerEnumFormatter](`https://github.com/scalapb/scalapb-json4s/blob/7277b4825ad12e661767f77451f7f1b8f51b3a97/src/main/scala/scalapb/json4s/JsonFormat.scala#L43). If we go this path, then this PR would go on the next minor version due to the incompatibility.